### DISCUSSION TITLE
How do I get INAV on Betafpv F405 Toothpick v5 (AIO) ?

### DISCUSSION THREAD

**Discussion by armnation** [iNavFlight/inav#9756] [Q&A]
Hello,
I'm pretty new to FPV world. I am using [BETAFPV F405 20A Toothpick](https://betafpv.com/products/f405-4s-20a-toothpick-brushless-flight-controller-v5-blheli_s-icm42688) as a flight controller. It comes preinstalled with betaflight.
I want to install INAV on this. I was unable to find this target in the target directory of the INAV. So, what are my options ?

1. do I have to create a separate port for this flight controller from scratch ? is there any tutorial for that ?
2. I was able to find [DAKEFPV F405](http://dakefpv.com/pd.jsp?id=27) in the INAV target directory, which looks almost similar to my BETAFPVF405 (only IMU is different). Can I just use this target without modification ? Or how do I modify it, assuming I'd need to add support for ICM42688 ?
3. Is there any other target that is more suitable for BETAFPV F405 Toothpick, which would need least amount of modification ?

Any kind of help is appreciated.
Thanks in advance.

**Comment by MrD-RC** [comment]
There is a policy for getting new hardware supported by INAV. You should contact BetaFPV and let them know that you would like to see n INAV target for that flight controller. The more people asking the manufacturers. The more likely that they will think of an INAV target in the future. 

https://github.com/iNavFlight/inav/blob/master/docs/policies/NEW_HARDWARE_POLICY.md

**Comment by armnation** [comment]
I have managed to develop a working firmware for the BETAFPVF405 Toothpick 20A v5 from its betaflight config.

Things that are working (that i was able to test):
- IMU
- RECEIVER (CRSF) + telemetry
- MOTORS spin

I do not have any sensors/camera at the moment to try it out. But I guess it'd work.
Should I create a PR for this ? (I'm not a manufacturer, and do not have a board to donate)

**Reply by evanmahendra** [reply]
Hello. Can i get that firmware bin files to help you testing with my environment? I will report it soon after testing.

**Reply by Bridgeburner250** [reply]
I know this was a long time ago but I would like to test that also. 🤙

**Reply by KTM1551** [reply]
I would also like the bin or hex files for INAV for the BetaFPV F405 AIO..? Thanks!!

I tried making it myself with the "build.h" script tool, but no matter what i do i cant get my PC to work properly with the Mingmw tool, or something similiar to that. 
It seems like i could use my RPi to do it since its a python script from my reasearch and what you need to do it on PC..

**Comment by sensei-hacker** [comment]
I see that BetaFPV apparently did a target for two of their other boards. You may want to email them and ask a) if they already made a target that they haven't shared with us yet or b) if they would be willing to help with a target.

**Reply by KTM1551** [reply]
Has anything happened with this? Do you know if anyone tried contacting BetaFPV about this? 

How could i make it by scratch myself? I have a spare AIO board and frame and i dont mind having to do some testing/re-flashing with corrections, if it means i can run INAV on my Pavo20Pro lol.. 

Plus then I would use this AIO on other builds as its fairly inexpensive, it's been known to actually support 4s, but rated for 3s, it has 20A ESC rating which is fine for most smaller builds as i tend to build light at the same time as building smaller drones. Its also very easy to get this AIO board from Beta directly or most of the time right off Amazon Prime.. 

Thanks!

**Reply by sensei-hacker** [reply]
To do it yourself, you could start with another F405 target, updating the pin definitions. For example:
https://github.com/iNavFlight/inav/tree/master/src/main/target/BOTWINGF405

Then build the new target as described in the appropriate file here:
https://github.com/iNavFlight/inav/tree/master/docs/development

Linux is probably the easiest platform to build on.
